usecases/mu/mpc/usecase.go

```diff
@@ -0,0 +1,605 @@
+//   - monitorFrequencyConfig: (optional) configuration parameters for MPC scenario, nil if not supported
+//
+// possible errors:
+//   - if required fields in parameters are unset
+func NewMPC(
+	localEntity spineapi.EntityLocalInterface,
+	eventCB api.EntityEventCallback,
+	monitorPowerConfig *MonitorPowerConfig,
+	monitorEnergyConfig *MonitorEnergyConfig,
+	monitorCurrentConfig *MonitorCurrentConfig,
+	monitorVoltageConfig *MonitorVoltageConfig,
+	monitorFrequencyConfig *MonitorFrequencyConfig,
+) (*MPC, error) {
+	if monitorPowerConfig == nil {
+		return nil, errors.New("the monitor power config for the MPC-Use-Case must not be nil")
+	}
+
+	validActorTypes := []model.UseCaseActorType{model.UseCaseActorTypeMonitoringAppliance}
+	useCaseScenarios := []api.UseCaseScenario{
+		{
+			Scenario:  model.UseCaseScenarioSupportType(1),
+			Mandatory: true,
+			ServerFeatures: []model.FeatureTypeType{
+				model.FeatureTypeTypeElectricalConnection,
+				model.FeatureTypeTypeMeasurement,
+			},
+		},
+	}
+
+	if monitorEnergyConfig != nil {
+		useCaseScenarios = append(useCaseScenarios, api.UseCaseScenario{
+			Scenario:  model.UseCaseScenarioSupportType(2),
+			Mandatory: false,
+			ServerFeatures: []model.FeatureTypeType{
+				model.FeatureTypeTypeElectricalConnection,
+				model.FeatureTypeTypeMeasurement,
+			},
+		})
+	}
+
+	if monitorCurrentConfig != nil {
+		useCaseScenarios = append(useCaseScenarios, api.UseCaseScenario{
+			Scenario:  model.UseCaseScenarioSupportType(3),
+			Mandatory: false,
+			ServerFeatures: []model.FeatureTypeType{
+				model.FeatureTypeTypeElectricalConnection,
+				model.FeatureTypeTypeMeasurement,
+			},
+		})
+	}
+
+	if monitorVoltageConfig != nil {
+		useCaseScenarios = append(useCaseScenarios, api.UseCaseScenario{
+			Scenario:  model.UseCaseScenarioSupportType(4),
+			Mandatory: false,
+			ServerFeatures: []model.FeatureTypeType{
+				model.FeatureTypeTypeElectricalConnection,
+				model.FeatureTypeTypeMeasurement,
+			},
+		})
+	}
+
+	if monitorFrequencyConfig != nil {
+		useCaseScenarios = append(useCaseScenarios, api.UseCaseScenario{
+			Scenario:  model.UseCaseScenarioSupportType(5),
+			Mandatory: false,
+			ServerFeatures: []model.FeatureTypeType{
+				model.FeatureTypeTypeElectricalConnection,
+				model.FeatureTypeTypeMeasurement,
+			},
+		})
+	}
+
+	u := usecase.NewUseCaseBase(
+		localEntity,
+		model.UseCaseActorTypeMonitoredUnit,
+		model.UseCaseNameTypeMonitoringOfPowerConsumption,
+		"1.0.0",
+		"release",
+		useCaseScenarios,
+		eventCB,
+		UseCaseSupportUpdate,
+		validActorTypes,
+		nil,
+		true,
+	)
+
+	uc := &MPC{
+		UseCaseBase:     u,
+		powerConfig:     monitorPowerConfig,
+		energyConfig:    monitorEnergyConfig,
+		currentConfig:   monitorCurrentConfig,
+		voltageConfig:   monitorVoltageConfig,
+		frequencyConfig: monitorFrequencyConfig,
+	}
+
+	_ = spine.Events.Subscribe(uc)
+
+	return uc, nil
+}
+
+func (e *MPC) AddFeatures() {
```

Comment:
This method is getting quite loooooooong. How about splitting it up into individual functions for each config?
Also I am still not sure if using panics is the right approach. Maybe `AddFeatures()` signature should be changed to `AddFeatures() error` overall ?

Also the code looks quite repetitive, and maybe there is a good way to shrink this down, and thus making it also easire to improve the test coverage for all error scenarios?

Reply:
We split the constituent measurement types into separate methods, and adjusted AddFeatures and AddUsecase to return error for all usecases